src/core/server/http/http_server.ts

```diff
@@ -167,66 +167,49 @@ export class HttpServer {
-        const validate = isSafeMethod(route.method) ? undefined : { payload: true };
         const { authRequired, tags, body = {}, timeout } = route.options;
         const { accepts: allow, maxBytes, output, parse } = body;
 
         const kibanaRouteOptions: KibanaRouteOptions = {
           xsrfRequired: route.options.xsrfRequired ?? !isSafeMethod(route.method),
         };
 
-        // To work around https://github.com/hapijs/hapi/issues/4122 until v20, set the socket
-        // timeout on the route to a fake timeout only when the payload timeout is specified.
-        // Within the onPreAuth lifecycle of the route itself, we'll override the timeout with the
-        // real socket timeout.
-        const fakeSocketTimeout = timeout?.payload ? timeout.payload + 1 : undefined;
-
-        this.server.route({
+        const routeOpts: ServerRoute = {
           handler: route.handler,
           method: route.method,
           path: route.path,
           options: {
             auth: this.getAuthOption(authRequired),
             app: kibanaRouteOptions,
-            ext: {
-              onPreAuth: {
-                method: (request, h) => {
-                  // At this point, the socket timeout has only been set to work-around the HapiJS bug.
-                  // We need to either set the real per-route timeout or use the default idle socket timeout
-                  if (timeout?.idleSocket) {
-                    request.raw.req.socket.setTimeout(timeout.idleSocket);
-                  } else if (fakeSocketTimeout) {
-                    // NodeJS uses a socket timeout of `0` to denote "no timeout"
-                    request.raw.req.socket.setTimeout(this.config!.socketTimeout ?? 0);
-                  }
-
-                  return h.continue;
-                },
-              },
-            },
             tags: tags ? Array.from(tags) : undefined,
-            // TODO: This 'validate' section can be removed once the legacy platform is completely removed.
-            // We are telling Hapi that NP routes can accept any payload, so that it can bypass the default
-            // validation applied in ./http_tools#getServerOptions
-            // (All NP routes are already required to specify their own validation in order to access the payload)
-            validate,
-            payload: [allow, maxBytes, output, parse, timeout?.payload].some(
-              (v) => typeof v !== 'undefined'
-            )
+            // @ts-expect-error Types are outdated and doesn't allow `payload.multipart` to be `true`
+            payload: [allow, maxBytes, output, parse, timeout?.payload].some((x) => x !== undefined)
               ? {
                   allow,
                   maxBytes,
                   output,
                   parse,
                   timeout: timeout?.payload,
+                  multipart: true, // TODO: Figure out what the default value should be
```

Comment:
@elastic/kibana-core I'm not sure if this is what we want. However, hapi v19 contain the following braking change:

> Change the route `options.payload.multipart` to `false` by default (hapijs/hapi#3920).

The question is if we want to change this to `true` across the board for _all_ routes, or if we want to be more selective about either which route it's applied to, or if we want to use the object-variant instead (see https://hapi.dev/api/?v=20.0.3#-routeoptionspayloadmultipart).

Reply:
I think it would make sense to change the default at one point, but it feels like unnecessary added risk for an upgrade. I think we should do something like that separately.

Reply:
Sounds good. I've removed the TODO then 👍

Reply:
created https://github.com/elastic/kibana/issues/86085